src/ZeroLog.Impl.Full/UnmanagedCache.cs

```diff
@@ -42,10 +41,15 @@ internal static void Register(Type unmanagedType)
         if (!typeof(ISpanFormattable).IsAssignableFrom(unmanagedType))
             throw new ArgumentException($"Not an {nameof(ISpanFormattable)} type: {unmanagedType}");
 
-        if (!TypeUtil.GetIsUnmanagedSlow(unmanagedType))
-            throw new ArgumentException($"Not an unmanaged type: {unmanagedType}");
+        _registerGuardedMethod.MakeGenericMethod(unmanagedType).Invoke(null, null);
+    }
 
-        _registerMethod.MakeGenericMethod(unmanagedType).Invoke(null, null);
+    private static void RegisterGuarded<T>(UnmanagedFormatterDelegate<T> formatter)
+        where T : unmanaged
```

Comment:
If we call `RegisterGuarded<T>` with a managed `T` through reflection, the `where T : unmanaged` constraint will be violated.

Not sure if that's a problem or not, but I think I prefer #48 because of that.